### PR TITLE
COMMISS.ENH: Splits get_orbit_data into other two better methods

### PR DESCRIPTION
**Pull request by jquentino**
Following a @murilobalves tip, I split get_orbit_data into two methods:
1 - get_orbit: stores orbit and  timestamp;
2 - get_data: stores devices data (i. e. current, rf frequency, acquisition rate...).

In addition, I also add new entries in `self.data` dict to store the quadrupoles waveform and the delta of tunes.

### CODE REVIEW

File: apsuite/commisslib/meas_bo_tune_by_bpm.py [hunk 1]
```diff
@@ -82,13 +82,13 @@ def configure_bpms(self):
...
-    def get_orbit_data(self, injection=False, external_trigger=False):
-        """Get orbit data from BPMs in TbT acquisition rate
-        BPMs must be configured to listen DigBO event and the DigBO
-        event must be in Injection mode.
+    def get_orbit(self, injection=False, external_trigger=False):
+        """Get orbit data from BPMs in TbT acquisition rate..
         If injection is True, then injection is turned on before the measure.
         If external_trigger is True, the event will listen a external trigger.
         """
+        if not hasattr(self, 'data'):
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_bo_tune_by_bpm.py:90]: why is this needed? For this type of object the `data` dictionary is always defined at initialization.

File: apsuite/commisslib/meas_bo_tune_by_bpm.py [hunk 2]
```diff
@@ -107,30 +107,45 @@ def get_orbit_data(self, injection=False, external_trigger=False):
...
         orbx, orby = bobpms.get_mturn_orbit()
         bobpms.cmd_mturn_acq_abort()
         trigbpm.delay = self.delay0
 
+        self.data['orbx'], self.data['orby'] = orbx, orby
+        self.data['timestamp'] = _time.time()
+
+    def get_data(self, delta=''):
+        """."""
+        if not hasattr(self, 'data'):
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_bo_tune_by_bpm.py:121]: same here.

File: apsuite/commisslib/meas_bo_tune_by_bpm.py [hunk 3]
```diff
@@ -107,30 +107,45 @@ def get_orbit_data(self, injection=False, external_trigger=False):
...
         data['bpms_nrsamples_pre'] = bpm0.acq_nrsamples_pre
         data['bpms_nrsamples_post'] = bpm0.acq_nrsamples_post
         data['bpms_trig_delay_raw'] = trigbpm.delay_raw
         data['bpms_switching_mode'] = csbpm.SwModes._fields[
                                         bpm0.switching_mode]
-        data['init_delay'] = self.delay0
-        self.data = data
+        data['qf_wave_form'] = PV('BO-Fam:PS-QF:Wfm-RB').get()
+        data['qd_wave_form'] = PV('BO-Fam:PS-QD:Wfm-RB').get()
+        if hasattr(self, 'delay0'):
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_bo_tune_by_bpm.py:145]: it is not a good practice to have a `delay0` attribute of the measurement class. If this variable reflects a parameter of the measurement, you could add it to the `Params` object, if it is a measured value, store it in the `data` dictionary. If it is a temporary variable, keep it as temporary.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

File: apsuite/commisslib/meas_bo_tune_by_bpm.py (hunk 1 above)
Comment (jquentino) [reply apsuite/commisslib/meas_bo_tune_by_bpm.py]: Thanks for the observation, I didn't figure out that. 
I removed these lines.

File: apsuite/commisslib/meas_bo_tune_by_bpm.py (hunk 3 above)
Comment (jquentino) [reply apsuite/commisslib/meas_bo_tune_by_bpm.py]: I leave it as a temporary variable. Thanks for the tip.

**Review by murilobalves — CHANGES_REQUESTED** [review]

File: apsuite/commisslib/meas_bo_tune_by_bpm.py [hunk 4]
```diff
@@ -105,41 +103,49 @@ def get_orbit_data(self, injection=False, external_trigger=False):
...
-        data['orbx'], data['orby'] = orbx, orby
         data['mt_acq_rate'] = csbpm.AcqChan._fields[bpm0.acq_channel]
         data['bpms_nrsamples_pre'] = bpm0.acq_nrsamples_pre
         data['bpms_nrsamples_post'] = bpm0.acq_nrsamples_post
         data['bpms_trig_delay_raw'] = trigbpm.delay_raw
         data['bpms_switching_mode'] = csbpm.SwModes._fields[
                                         bpm0.switching_mode]
-        data['init_delay'] = self.delay0
-        self.data = data
+        data['qf_wave_form'] = PV('BO-Fam:PS-QF:Wfm-RB').get()
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_bo_tune_by_bpm.py:139]: This is not a good practice. You should create the PVs you need at the object initialization, together with the devices creation.  Add the PVs to the dictionary `self.pvs` which is already available for this type of object. All PVs added to this dictionary will be checked when the method `self.connected` or `self.wait_for_connection` is called.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

File: apsuite/commisslib/meas_bo_tune_by_bpm.py [hunk 5]
```diff
@@ -107,30 +107,45 @@ def get_orbit_data(self, injection=False, external_trigger=False):
...
+            raise AssertionError(
+                f'Problem waiting BPMs update. Error code: {ret:d}')
         orbx, orby = bobpms.get_mturn_orbit()
         bobpms.cmd_mturn_acq_abort()
         trigbpm.delay = self.delay0
 
+        self.data['orbx'], self.data['orby'] = orbx, orby
+        self.data['timestamp'] = _time.time()
+
+    def get_data(self, delta=''):
```
[the remark this answers is not in the document]
Comment (jquentino) [reply apsuite/commisslib/meas_bo_tune_by_bpm.py]: Sorry for the misunderstanding. Now I implemented it in this way, thanks!

File: apsuite/commisslib/meas_bo_tune_by_bpm.py (hunk 4 above)
Comment (jquentino) [reply apsuite/commisslib/meas_bo_tune_by_bpm.py]: Ok, done!

**Review by fernandohds564 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Review by murilobalves — APPROVED** [review]

**Review by xresende — APPROVED** [review]
